### PR TITLE
Bug-fixes and cleanups related to data insertions

### PR DESCRIPTION
**Pull request by romaninsh**
- [x] Merge Please merge #43, #45 and #46 first.

### CODE REVIEW

File: src/Field_Many.php [hunk 1]
```diff
@@ -62,22 +66,27 @@ public function init()
         $this->_init();
     }
 
-    protected function getModel()
+    protected function getModel($defaults = [])
```
Comment (skondakov) [inline src/Field_Many.php]: Nice option for default settings here

File: src/Model.php [hunk 2]
```diff
@@ -594,12 +630,38 @@ public function tryLoadAny()
...
+    public function loadBy($field, $value)
+    {
+        $this->addCondition($field, $value);
+        try {
+            $this->loadAny();
+        } catch (\Exception $e) {
+            array_pop($this->conditions);
+            throw $e;
+        }
+        array_pop($this->conditions);
```
Comment (skondakov) [inline src/Model.php]: Maybe this block here can be refactored with try .. catch .. finally

File: src/Model.php [hunk 3]
```diff
@@ -594,12 +630,38 @@ public function tryLoadAny()
...
+    public function tryLoadBy($field, $value)
+    {
+        $this->addCondition($field, $value);
+        try {
+            $this->tryLoadAny();
+        } catch (\Exception $e) {
+            array_pop($this->conditions);
+            throw $e;
+        }
+        array_pop($this->conditions);
```
Comment (skondakov) [inline src/Model.php]: Same here

**Comment by skondakov** [comment]
All looking good except the try..catch..finally notes. Good to go IMO! Great work!

File: src/Model.php (hunk 2 above)
Comment (romaninsh) [inline src/Model.php]: Well noted. Added issue #51 to deal with this.

File: src/Model.php (hunk 3 above)
Comment (romaninsh) [inline src/Model.php]: #51 will deal with it.

**Pull request merged**